diff --git a/Dockerfile b/Dockerfile
index b08b7d2..b64472e 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM debian:bookworm-slim AS builder
+FROM debian:bookworm-slim@sha256:63a496b5d3b99214b39f5ed70eb71a61e590a77979c79cbee4faf991f8c0783e AS builder
 
 WORKDIR /wanco
 
@@ -27,11 +27,12 @@ RUN \
 ENV LLVM_SYS_170_PREFIX=/usr/lib/llvm-17
 
 # install Rust
-RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | \
+    sh -s -- -y --profile minimal --default-toolchain 1.97.1
 ENV PATH="/root/.cargo/bin:/usr/lib/llvm-17/bin:${PATH}"
 
 # hyperfine for benchmarking
-RUN cargo install hyperfine
+RUN cargo install hyperfine --locked --version 1.20.0
 
 COPY . /wanco
 
