diff --git a/wanco/src/compile/control.rs b/wanco/src/compile/control.rs
index 4c9e9ae..bf61223 100644
--- a/wanco/src/compile/control.rs
+++ b/wanco/src/compile/control.rs
@@ -18,6 +18,8 @@ use super::{
     },
 };
 
+const POST_IMPORT_CHECKPOINT_FLAG: u32 = 1 << 31;
+
 /// Holds the state of if-else.
 #[derive(Eq, PartialEq, Debug)]
 pub enum IfElseState {
@@ -610,6 +612,14 @@ pub fn gen_call<'a>(
                 .expect("fail translate call_site"),
         );
     }
+    if ctx.config.enable_cr && callee_function_index < ctx.num_imports {
+        let call_op = ctx.current_op.expect("call should have an operator index");
+        assert_eq!(call_op & POST_IMPORT_CHECKPOINT_FLAG, 0);
+        ctx.current_op = Some(call_op | POST_IMPORT_CHECKPOINT_FLAG);
+        gen_migration_point(ctx, exec_env_ptr, locals)?;
+        ctx.current_op = Some(call_op);
+        ctx.num_migration_points += 1;
+    }
     Ok(())
 }
 
